forge/ee/lib/tables/drivers/postgres-localfs.js

```diff
@@ -0,0 +1,241 @@
+const pg = require('pg')
+
+const { generatePassword } = require('../../../../lib/userTeam')
+
+let adminClient
+
+module.exports = {
+    init: async function (app, options) {
+        this._app = app
+        this._options = options || {}
+        if (!options.database) {
+            throw new Error('Postgres LocalFS driver requires database options to be provided')
+        }
+        adminClient = new pg.Client(options.database || {})
+        adminClient.on('error', (err) => {
+            this._app.log.error('Postgres LocalFS driver error:', err)
+        })
+        try {
+            await adminClient.connect()
+        } catch (err) {
+            app.log.error('Failed to connect to Postgres:', err)
+        }
+        app.log.info('Postgres LocalFS driver initialized')
+    },
+    shutdown: async function () {
+        try {
+            this._app.log.info('Shutting down Postgres LocalFS driver')
+            await adminClient.end()
+        } catch (err) {
+            this._app.log.debug('Error shutting down Postgres LocalFS driver:', err)
+        }
+    },
+    getDatabases: async function (team) {
+        const tables = await this._app.db.models.Table.byTeamId(team.id)
+        if (tables && tables.length > 0) {
+            return tables
+        } else {
+            return []
+        }
+    },
+    getDatabase: async function (team, databaseId) {
+        const table = await this._app.db.models.Table.byId(team.id, databaseId)
+        if (table) {
+            return table
+        } else {
+            throw new Error(`Database ${databaseId} for team ${team.hashid} does not exist`)
+        }
+    },
+    createDatabase: async function (team, name) {
+        const existing = await this._app.db.models.Table.byTeamId(team.id)
+        // Will need removing when we support multiple databases per team
+        if (existing && existing.length > 0) {
+            throw new Error('Database already exists')
+        }
+        const res = await adminClient.query('SELECT datname FROM pg_database WHERE datistemplate = false AND datname = $1', [team.hashid])
+        if (res.rows.length > 0) {
+            throw new Error('Database already exists')
+        } else {
+            await adminClient.query(`CREATE DATABASE "${team.hashid}"`)
+            await adminClient.query(`REVOKE connect ON DATABASE "${team.hashid}" FROM PUBLIC;`)
+            const options = {
+                host: this._options.database.host,
+                port: this._options.database.port,
+                ssl: this._options.database.ssl,
+                database: team.hashid,
+                user: this._options.database.user,
+                password: this._options.database.password
+            }
+            const teamClient = new pg.Client(options)
```

Comment:
This same code is duplicated in 4 places.
I feel this should be extracted to a reusable/testable function.
Additionally, being a shared lib function would permit us to mock `pg.create` and do proper unit testing on the drivers